import-scripts/clear-persistence-caches-for-changed-studies.sh

```diff
@@ -0,0 +1,450 @@
+# usage : search_for_studies_with_new_modifications_in_db db_label
+function search_for_studies_with_new_modifications_in_db() {
+    db="$1"
+    tmp_study_subdir="$TMP_DIR_ROOT/$db"
+    create_directory_if_necessary "$tmp_study_subdir"
+    previous_study_info_from_db_filepath="$TMP_DIR_ROOT/${db}_study_info"
+    new_study_info_from_db_filepath="$TMP_DIR_ROOT/${db}_study_info_new"
+    if ! [ -f "$previous_study_info_from_db_filepath" ] ; then
+        # if we have lost the previous study info, we cannot tell what has been modified. Prime the study info and trigger a full cache reset
+        write_current_study_list_from_db "$db" "$previous_study_info_from_db_filepath"
+        trigger_db_as_needing_reset "$db"
+    fi
+    write_current_study_list_from_db "$db" "$new_study_info_from_db_filepath"
+    find_studies_deleted_added_or_changed "$previous_study_info_from_db_filepath" "$new_study_info_from_db_filepath"
+}
+
+# creates the trigger file for a study cache reset
+# usage : trigger_study_as_needing_reset db_label study_id
+function trigger_study_as_needing_reset() {
+    db="$1"
+    study_id="$2"
+    tmp_study_subdir="$TMP_DIR_ROOT/$db"
+    study_reset_needed_filepath="$tmp_study_subdir/$study_id.$STUDY_CACHE_INVALIDATION_NEEDED_SUFFIX"
+    touch "$study_reset_needed_filepath"
+}
+
+# extracts the cache api key from a configmap yaml file. The key is stored in a global associative array
+# usage : extract_cache_clearing_api_key db_label
+function extract_cache_clearing_api_key() {
+    db="$1"
+    extractPropertiesFromConfigMapFile ${PORTAL_DB_TO_CONFIG_MAP_FILE[$db]} CACHE_API_KEY
+    portal_db_to_cache_api_key[$db]="${extracted_properties[CACHE_API_KEY]}"
+}
+
+# replaces the previous study list snapshot with the latest (called after any necessary triggers have been created)
+# usage : overwrite_previous_with_new_study_info db_label
+function overwrite_previous_with_new_study_info() {
+    db="$1"
+    tmp_study_subdir="$TMP_DIR_ROOT/$db"
+    previous_study_info_from_db_filepath="$TMP_DIR_ROOT/${db}_study_info"
+    new_study_info_from_db_filepath="$TMP_DIR_ROOT/${db}_study_info_new"
+    if [ -f "$new_study_info_from_db_filepath" ] ; then
+        mv "$new_study_info_from_db_filepath" "$previous_study_info_from_db_filepath"
+    fi
+}
+
+# phase one top level logic - sets up / cleans up / creates the task triggers and status directories/content for every managed database
+# also has logic to override study cache reset with a full cache reset when task backlog gets stale
+function detect_and_maintain_needed_actions_for_databases() {
+    create_directory_if_necessary "$TMP_DIR_ROOT"
+    for portal_db in $PORTAL_DB_LIST ; do
+        clear_completed_triggers "$portal_db"
+        if ! database_is_triggered_as_needing_reset "$portal_db" ; then
+            if any_trigger_has_exceeded_maximum_wait "$portal_db"; then
+                trigger_db_as_needing_reset "$portal_db"
```

Comment:
continue to the next loop iteration here.

Reply:
Done